src/pool/mempool.rs

```diff
@@ -58,42 +71,77 @@ impl AccountManager {
+
+                    // Convert the optional Ethereum block ID to a Starknet block ID.
+                    let starknet_block_id = eth_provider.to_starknet_block_id(Some(BlockId::default())).await?;
+
+                    // Create a new `ERC20Reader` instance for the Starknet native token
+                    let eth_contract = ERC20Reader::new(*STARKNET_NATIVE_TOKEN, eth_provider.starknet_provider());
+
+                    // Call the `balanceOf` method on the contract for the given address and block ID, awaiting the result
+                    let span = tracing::span!(tracing::Level::INFO, "sn::balance");
+                    let res = eth_contract.balanceOf(address).block_id(starknet_block_id).call().instrument(span).await;
+
+                    if contract_not_found(&res) || class_hash_not_declared(&res) {
+                        return Err(eyre::eyre!("Contract not found or class hash not declared"));
+                    }
+
+                    // Otherwise, extract the balance from the result, converting any errors to ExecutionError
+                    let balance = res.map_err(ExecutionError::from)?.balance;
+
+                    // Convert the low and high parts of the balance to U256
+                    let low: U256 = into_via_wrapper!(balance.low);
+                    let high: U256 = into_via_wrapper!(balance.high);
+
+                    // Combine the low and high parts to form the final balance and return it
+                    let balance = low + (high << 128);
+
+                    if oz_account.current_tx_hash.is_none() && balance > U256::from(u128::pow(10, 18)) {
+                        Self::process_transaction(address, eth_provider);
+                    }
                 }
 
                 tokio::time::sleep(Duration::from_secs(1)).await;
             }
         });
+
+        rt_handle.spawn(async {
+            if let Err(e) = handle.await {
+                tracing::error!("Error in spawned task: {:?}", e);
+            }
+        });
+
+        Ok(())
     }
 
-    fn process_transaction<SP>(address: &Felt, eth_provider: &EthDataProvider<SP>)
+    fn process_transaction<SP>(_address: &Felt, eth_provider: &EthDataProvider<SP>)
     where
         SP: starknet::providers::Provider + Send + Sync + 'static,
     {
-        let balance = Self::check_balance(address);
+        let best_hashes =
+            eth_provider.mempool.as_ref().unwrap().best_transactions().map(|x| *x.hash()).collect::<Vec<_>>();
 
-        if balance > Felt::ONE {
-            let best_hashes =
-                eth_provider.mempool.as_ref().unwrap().best_transactions().map(|x| *x.hash()).collect::<Vec<_>>();
-
-            if let Some(best_hash) = best_hashes.first() {
-                eth_provider.mempool.as_ref().unwrap().remove_transactions(vec![*best_hash]);
-            }
+        if let Some(best_hash) = best_hashes.first() {
+            eth_provider.mempool.as_ref().unwrap().remove_transactions(vec![*best_hash]);
```

Comment:
so for now we just remove the transaction and don't send it right?

Reply:
Yes exact and in a follow up we will remove the transaction and send it via the OpenZeppelin account